libraries/c_sdk/standard/mqtt/src/iot_mqtt_subscription.c

```diff
@@ -416,92 +406,106 @@ void _IotMqtt_InvokeSubscriptionCallback( _mqttConnection_t * pMqttConnection,
         .exactMatchOnly  = false
     };
 
-    /* Prevent any other thread from modifying the subscription list while this
+    contextIndex = _IotMqtt_getContextIndexFromConnection( pMqttConnection );
+
+    /* Prevent any other thread from modifying the subscription array while this
      * function is searching. */
-    IotMutex_Lock( &( pMqttConnection->subscriptionMutex ) );
+    if( IotMutex_Take( &( connToContext[ contextIndex ].subscriptionMutex ) ) == false )
+    {
+        /* Fail to take context mutex due to timeout. */
+        mutexStatus = false;
```

Comment:
can this be mutexStatus  = IotMutex_Take( &( connToContext[ contextIndex ].subscriptionMutex ) );?

Reply:
Addressed